### PR TITLE
Cherry-pick #16306 to 7.x: Enable keystore for autodiscover static configuration

### PR DESCRIPTION
**Pull request by ChrsMark**
Cherry-pick of PR #16306 to 7.x branch. Original message: 

Close https://github.com/elastic/beats/issues/12597.

## What this PR does
This PR makes use of keystore for containers/pods that are autodiscovered with static configurations only. Hint based configuration should not have access to the keystore for security reasons.

## How to test it manually

1. Create the keystore to store password for REDIS:
```console
./metricbeat keystore create                                                                                 
Created metricbeat keystore
./metricbeat keystore add REDIS_PASSWORD                                                                      
Enter value for REDIS_PASSWORD: 
Successfully updated the keystore
```
1. Enable `autodiscover` with static template:
```
metricbeat.autodiscover:
  providers:
    - type: docker
      templates:
        - condition:
            contains:
              docker.container.image: redis
          config:
            - module: redis
              metricsets: ["info", "keyspace"]
              hosts: ["${data.host}:6379"]
              password: ${REDIS_PASSWORD}
```
2. Start metricbeat
3. Start a password redis server in a container to be autodiscovred:
```
docker run -p 6379:6379 \
--entrypoint "redis-server" redis --appendonly yes --requirepass 'passpass'
```
Check that REDIS metrics are successfully collected.
Try to start REDIS with a different password so as to make Metricbeat fail.

Now check that hints based autodiscovered containers have not access to the keystore:
1. Configure hint's based autodiscover:
```
metricbeat.autodiscover:
  providers:
    - type: docker
      hints.enabled: true
```
2. Check that hints have no access to the keystore:
```
docker run -p 6379:6379 \
-l co.elastic.metrics/module=redis \
-l co.elastic.metrics/metricsets=info \
-l co.elastic.metrics/password='${REDIS_PASSWORD}' \
-l co.elastic.metrics/hosts='${data.host}:6379' \
--entrypoint "redis-server" redis --appendonly yes --requirepass 'passpass'
```
You should see Metribeat failing to access REDIS since the password is not retrievable.

cc: @exekias

### CODE REVIEW

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -263,6 +263,8 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Add `replace` processor for replacing string values of fields. {pull}17342[17342]
 - Add `urldecode` processor to for decoding URL-encoded fields. {pull}17505[17505]
 - Add support for AWS IAM `role_arn` in credentials config. {pull}17658[17658] {issue}12464[12464]
+- Add keystore support for autodiscover static configurations. {pull]16306[16306]
+- Add Kerberos support to Elasticsearch output. {pull}17927[17927]
```
Comment (exekias) [inline CHANGELOG.next.asciidoc:267]: leftover here

**Comment by ChrsMark** [comment]
It think I will include the "reverted" changes for the broken docs after https://github.com/elastic/beats/pull/18097 is merged to master.

**Comment by ChrsMark** [comment]
Removed docs changes that were reverted on #18097 .

**Comment by ChrsMark** [comment]
jenkins, test this please

**Pull request merged**